src/Nethermind/Nethermind.Core/BlockInfo.cs

```diff
@@ -32,9 +39,18 @@ public BlockInfo(Keccak blockHash, in UInt256 totalDifficulty)
         public bool WasProcessed { get; set; }
         
         public Keccak BlockHash { get; }
-        
-        public bool IsFinalized { get; set; }
-        
+
+        public bool IsFinalized
+        {
+            get => FinalizationStatus == FinalizationStatus.Finalized;
+            set
+            {
+                FinalizationStatus = value ? FinalizationStatus.Finalized : FinalizationStatus.None;
+            }
```

Comment:
If it's all depend on `FinalizationStatus`, I think `IsFinalized`should become a _getter_ only ?